src/charm.py

```diff
@@ -233,21 +296,33 @@ def configure_kube_ovn(self):
             self.apply_crds()
             self.apply_ovn(registry)
             self.apply_kube_ovn(service_cidr, registry)
+            self.apply_speakers(registry)
 
             if self.stored.pod_restart_needed:
                 self.wait_for_ovn_central()
                 self.wait_for_kube_ovn_controller()
                 self.wait_for_kube_ovn_cni()
+                self.wait_for_speakers()
                 self.restart_pods()
+
         except CalledProcessError:
             # Likely the Kubernetes API is unavailable. Log the exception in
-            # case it it something else, and let the caller know we failed.
+            # case it is something else, and let the caller know we failed.
             log.error(traceback.format_exc())
             self.unit.status = WaitingStatus("Waiting to retry configuring Kube-OVN")
             return
 
         self.stored.kube_ovn_configured = True
 
+    def apply_speakers(self, registry):
+        self.remove_speakers()
+        if self.model.config["bgp-speakers"]:
+            speaker_config_list = list(
+                yaml.safe_load(self.model.config["bgp-speakers"])
```

Comment:
i kinda wish there was some schema parsing here.  check out 

[pydantic](https://pydantic-docs.helpmanual.io/) if you're interested:

```python
from pydantic import BaseModel, Field, IPvAnyAddress

class SpeakerConfig(BaseModel):
    name: str
    node_selector: str = Field(alias="node-selector")
    neighbor_address: IPvAnyAddress = Field[alias="neighbor-address")
    neighbor_as: int = Field(alias="neighbor-as", gt=0, lt=65536)
    cluster_as: int = Field(alias='cluster-as', gt=0, lt=65536)
    announce_cluster_ip: bool = Field(alias="announce-cluster-ip")
    log_level: int = Field(alias="log-level", default=2)
...

    raw = list(yaml.safe_load(self.model.config["bgp-speakers"])
    parsed = [SpeakerConfig(**d) for d in raw]
    
```


since i don't really know the semantics of the schema i've probably messed something up.  But i like how pydantic makes objects for you from schema'd data and gives you nice exceptions to catch when you mess up

Reply:
yeah this would be a good addition! will add it to the PR